nova/compute/manager.py

```diff
@@ -2232,6 +2232,9 @@ def _build_resources(self, context, instance, requested_networks,
                 self.driver.clean_networks_preparation(instance, network_info)
             raise exception.BuildAbortException(instance_uuid=instance.uuid,
                     reason=e.format_message())
+        except exception.PortLimitExceeded as e:
```

Comment:
As discussed in slack, please add the `QuotaError` exception to the above except clause, so that any network stuff created will be cleaned.